dashboard/test/serializers/api/v1/pd/application_serializer_test.rb

```diff
@@ -9,7 +9,7 @@ class Api::V1::Pd::ApplicationSerializerTest < ::ActionController::TestCase
   test 'Invalid application data does not break serializer result' do
     # Application data could have more than 1 invalid fields
     app_data = build :pd_teacher_application_hash, course: :csp, school: 'invalid code'
-    app = create :pd_teacher_application, form_data_hash: app_data
+    app = build :pd_teacher_application, form_data_hash: app_data
```

Comment:
I went back and forth on this ... `create` throws a validation error now because we run `update_user_school_info!` before_validation now (it used to only be called from `on_successful_create` which is only called from the controller). I think it's good a validation error is thrown, but then this test fails. I'm not confident what this test is trying to ensure, so it's possible I'm defeating the purpose of it by doing this. Please chime in if you have strong opinions.

Reply:
TL;DR: you can leave this as `build` :-)

`build` does not appear to run `before_save`:
* https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories
* https://stackoverflow.com/questions/14098031/whats-the-difference-between-the-build-and-create-methods-in-factorygirl/18003276#18003276

when I change `build` to `create` here, the unit test fails with:
```
ERROR["test_Invalid_application_data_does_not_break_serializer_result", "Api::V1::Pd::ApplicationSerializerTest", 18.92780400067568]
 test_Invalid_application_data_does_not_break_serializer_result#Api::V1::Pd::ApplicationSerializerTest (18.93s)
NoMethodError:         NoMethodError: undefined method `school_type' for nil:NilClass
            app/models/school_info.rb:111:in `sync_from_schools'
            lib/school_info_deduplicator.rb:20:in `get_duplicate_school_info'
            app/models/pd/application/teacher_application.rb:111:in `update_user_school_info!'
            test/serializers/api/v1/pd/application_serializer_test.rb:12:in `block in <class:ApplicationSerializerTest>'
```

but... it looks like the test is saying that it wants to test the case where the model is invalid, and make sure that this case doesn't break the serializer. I'm not quite sure why this is important to test, but your change seems to be in the spirit of what it is trying to test, so I think what you have here looks good.

Reply:
Whoops! Meant to say `create` also runs `before_save`, but yes, everything you wrote makes sense. Sounds good!

Reply:
ohh, that makes sense. thanks!